dockstore-webservice/src/main/java/io/dockstore/webservice/resources/proposedGA4GH/ToolsApiExtendedServiceImpl.java

```diff
@@ -662,6 +662,7 @@ public Response updateAITopic(String id, UpdateAITopicRequest updateAITopicReque
             () -> new CustomWebApplicationException(VERSION_NOT_FOUND_ERROR, HttpStatus.SC_NOT_FOUND));
 
         entry.setTopicAI(updateAITopicRequest.getAiTopic());
+        entry.setApprovedAITopic(false);
```

Comment:
I wonder, if the user has already selected the AI topic and implicitly approved it, should we reject the attempt to change it, or something like that?  This would help to prevent us from accidentally overriding the user's previous action.  Of course, we would tweak the topic updater support app to perform the same check, this would be a backup.

Reply:
Perhaps I should perform a check to see if the new AI topic is different and only reset the `approvedAITopic` field to false if it is? That would prevent us from accidentally resetting the user's approval if the topic has not changed.

If the AI topic has actually changed, I think it will need another approval from the user because the content has changed

Reply:
Hmm, some random obs:

* Per my understanding of ChatGPT/etc, the generated topic will vary each time, even if we submit the same input.
* It'd be a bummer for the user to have to keep selecting/approving generated topics, even after trivial changes to the entry.
* We do want the user to know about and use a newly-generated topic, if the entry has changed enough that the previously-generated topic is obsolete.

Maybe before generation, we could ask ChatGPT if the existing topic still applies?  Or, after generating a new topic, ask ChatGPT if the new and previous topics mean the same thing, and not update if they are semantically the same?

Reply:
(the above suggests that the we do any necessary checks in the topic generator, and that the webservice logic you've implemented is good.)

Reply:
> It'd be a bummer for the user to have to keep selecting/approving generated topics, even after trivial changes to the entry.

Agreed, I added an additional check in the webservice so it only resets the approval if the topic has changed

> We do want the user to know about and use a newly-generated topic, if the entry has changed enough that the previously-generated topic is obsolete.

I wonder if we should even be re-generating AI topics if they've already approved of it. Maybe we allow them to choose if they want it regenerated? That would probably require some extra work though depending if we want instant re-generation and the 1.16 priority is likely generating topics for old and new entries without any topics.

> Maybe before generation, we could ask ChatGPT if the existing topic still applies? Or, after generating a new topic, ask ChatGPT if the new and previous topics mean the same thing, and not update if they are semantically the same?

Good idea, I created a ticket for that https://ucsc-cgl.atlassian.net/browse/SEAB-6590